beacon-chain/p2p/testing/p2p.go

```diff
@@ -101,6 +101,7 @@ func (p *TestP2P) ReceiveRPC(topic string, msg proto.Message) {
 
 	n, err := p.Encoding().EncodeWithMaxLength(s, msg)
 	if err != nil {
+		_ = s.Reset()
```

Comment:
This doesn't seem appropriate. Why ignore stream.Reset errors?

Reply:
1. They're not useful. We only call Reset when we run into some other more relevant error. At that point, we don't really care about the stream (other than freeing resources).
2. Reset is guaranteed to free resources, even if it returns an error.
3. No stream multiplexer actually returns an error on reset. We return errors for completeness, but if 
I were to redesign these interfaces from scratch, I probably wouldn't.